File: src/main/java/kr/co/lunatalk/domain/member/repository/MemberRepository.java
package kr.co.lunatalk.domain.member.repository;

import kr.co.lunatalk.domain.member.domain.Member;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface MemberRepository extends JpaRepository<Member, Long>, MemberRepositoryCustom {
	public Optional<Member> findByUsername(String username);
}

File: src/main/kotlin/kr/co/lunatalk/domain/member/repository/MemberRepository.kt
package kr.co.lunatalk.domain.member.repository

import kr.co.lunatalk.domain.member.domain.Member
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface MemberRepository : JpaRepository<Member, Long>, MemberRepositoryCustom {
    fun findByUsername(username: String): Optional<Member>
}
